sql/src/main/java/org/apache/druid/sql/calcite/schema/DruidCalciteSchema.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.sql.calcite.schema;
+
+import org.apache.calcite.schema.Schema;
+
+/**
+ * A calcite schema that has a name which it should be registered to.
+ */
+public interface DruidCalciteSchema extends Schema
```

Comment:
Do you think it makes sense to make this a SchemaProvider and have it return a Schema instead of being one? It always seems weird to me to extend other libraries' interfaces and add methods to them.

Reply:
Yeah I was being lazy and extended a libraries' interface. I've changed it to use composition instead since I need the name and the Schema .